integration/cluster.go

```diff
@@ -734,6 +734,52 @@ func (m *member) Metric(metricName string) (string, error) {
 	return "", nil
 }
 
+func (m *member) WaitLeaderChange(t *testing.T) uint64 {
+	oldLead := m.s.Lead()
+	timeout := m.s.Cfg.ReqTimeout()
+
+	donec, stopc := make(chan struct{}), make(chan struct{})
+	go func() {
+		defer close(donec)
+		for m.s.Lead() == oldLead {
+			select {
+			case <-stopc:
+				t.Fatalf("%s expected to leader change from %s (time-out %v)", m.s.ID(), types.ID(oldLead), timeout)
+			case <-time.After(time.Duration(m.s.Cfg.ElectionTicks) * time.Millisecond):
+			}
+		}
+	}()
+	select {
+	case <-donec:
+	case <-time.After(timeout):
+		close(stopc)
+	}
+
+	return m.s.Lead()
+}
+
+func (m *member) PauseSend(t *testing.T, toMember *member) {
```

Comment:
are these functions necessary? just call m.s.CutPeer/MendPeer directly? The send part is sort of misleading because it's also rejecting messages send from the peer too.